app/src/main/java/org/mozilla/fenix/utils/Settings.kt

```diff
@@ -465,13 +465,16 @@ class Settings(private val appContext: Context) : PreferencesHolder {
     )
 
     /**
-     * Caches the last known "is default browser" state when the app was paused.
-     * For an up to do date state use `isDefaultBrowser` instead.
+     * Declared as a function for performance purposes. When this class is first initialized, the
+     * isDefaultBrowser function takes approx. 30-40ms.
```

Comment:
**praise:** Nice – this function deserves a comment to explain why it exists and I appreciate that you documented the previous performance problem.

**suggestion (non-blocking):** but I have some thoughts on the comment:
- `isDefaultBrowser` no longer takes that time _when the class is first initialized_ – can we update the comment appropriately?
- it takes 30-40ms on which device?
- this comment actually feels like it belongs on `isDefaultBrowser`. Should we move it? And perhaps this would need it's own comment explaining why this should exist